disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -15,3 +17,10 @@ vectorOfSize min' max' gen =
 chooseSize :: Int -> Int -> Gen Int
 chooseSize min' max' =
   sized (return . min max' . max min')
+
+-- | from a generator return a generator that will generate Nothing
+--   half the time and Just half the time
+maybeGen :: Gen a -> Gen (Maybe a)
+maybeGen g = do
+  b <- elements [True, False]
```

Comment:
`oneof [Just <$> g, pure Nothing]`

The other thing to consider is whether you have a disproportionate frequency - should `Just` be more common than `Nothing`? This is what the real version [does](https://hackage.haskell.org/package/QuickCheck-2.8.1/docs/src/Test-QuickCheck-Arbitrary.html#Arbitrary):

    arbitrary = frequency [(1, return Nothing), (3, liftM Just arbitrary)]

Same with Scalacheck:

https://github.com/rickynils/scalacheck/blob/master/src/main/scala/org/scalacheck/Arbitrary.scala#L282